rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -115,11 +187,25 @@ module Node {
     override string toString() { result = node.toString() }
   }
 
-  final class ReturnNode extends NaNode {
-    RustDataFlow::ReturnKind getKind() { none() }
+  /** A data flow node that represents a value returned by a callable. */
+  final class ReturnNode extends ExprNode {
+    ReturnNode() { this.getCfgNode().getASuccessor() instanceof ExitCfgNode }
```

Comment:
I think this should just be `ReturnExpr`s.

Reply:
Ok, so we don't want to include implicit returns like `fn foo() { 12 }`?

Reply:
Right, we need to include that as well.

Reply:
I've reinstated the old implementation again :). If we want to include implicit returns then I think just looking at edges into the exit node handles all cases.
- `return` goes directly to the exit node
- In the absence of explicit returns, the `BlockExpr` is last both in a function and in a closure and it lets the final expression flow through if there is no `;` at the end.
- For a lambda without braces `|a| a + 1` the single expression has an edge to the exit node.